extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -0,0 +1,94 @@
+# KHR\_materials\_clearcoat
+
+## Khronos 3D Formats Working Group
+
+* Alexey Knyazev
+* Don McCurdy
+* Ed Mackey
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)
+* Romain Guy
+* Tobias Haeussler
+
+## Acknowledgments
+
+* TODO
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+TODO  
+
+## Extending Materials
+
+The PBR clearcoat materials are defined by adding the `KHR_materials_clearcoat` extension to any glTF material. 
+For example, the following defines a material like varnish using clearcoat parameters.
+
+```json
+{
+    "materials": [
+        {
+            "name": "varnish",
+            "extensions": {
+                "KHR_materials_clearcoat": {
+                    "clearcoatFactor": 1.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Clearcoat
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [appendix](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatFactor**               | `number`                                                                        | The clearcoat layer intensity.         | No, default: `0.0`   |
+|**clearcoatTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer intensity texture. | No                   |
+|**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
+|**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
+|**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
+  
+The clearcoat formula `f_clearcoat` is the same as the specular term from the Metallic-Roughness material, using F0 equal 0.04.
+The following abstract code describes how the base and clearcoat layers should be blended together:  
+  
+```
+blendFactor = clearcoatFactor * fresnel(0.04, NdotV)
+f = (f_emissive + f_diffuse) * (1.0 - blendFactor) + mix(f_specular, f_clearcoat, blendFactor)
```

Comment:
Can this formula be simplified as follows?  Is this the same formula?

```
f = mix(f_emissive + f_diffuse + f_specular, f_clearcoat, blendFactor)
```

Reply:
The formula I provided is from Substance Painter and they are just blending the specular part, which does makes sense from my point of view.
Where do you have the formula from?

Reply:
It's the same formula, just refactored.  Recall that:

```
mix(x, y, a) = x * (1.0 - a) + y*a
```

Given the left side of `mix` is already multiplied by `1.0 - blendFactor`, I don't think we need to show it being multiplied separately.

Looking at Substance Painter's code, I see why they did it that way: Their shader needed diffuse, emissive, and specular outputs as separate values, not added together.  Perhaps we could show it as separate values too, but if we're showing it all added together then it may as well all go into the mix statement.